src/components/Modals/Snaps/Snaps.tsx

```diff
@@ -0,0 +1,94 @@
+import {
+  Box,
+  Button,
+  Checkbox,
+  Heading,
+  HStack,
+  Image,
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalOverlay,
+  Text,
+} from '@chakra-ui/react'
+import { KnownChainIds } from '@shapeshiftoss/types'
+import { useCallback, useMemo } from 'react'
+import { useTranslate } from 'react-polyglot'
+import { enableShapeShiftSnap } from 'utils/snaps'
+import { getChainAdapterManager } from 'context/PluginProvider/chainAdapterSingleton'
+import { useFeatureFlag } from 'hooks/useFeatureFlag/useFeatureFlag'
+import { useModal } from 'hooks/useModal/useModal'
+import { useToggle } from 'hooks/useToggle/useToggle'
+import { isSome } from 'lib/utils'
+import { preferences } from 'state/slices/preferencesSlice/preferencesSlice'
+import { selectAssetById } from 'state/slices/selectors'
+import { store } from 'state/store'
+
+export const Snaps = () => {
+  const { close, isOpen } = useModal('snaps')
+  const isSnapsEnabled = useFeatureFlag('Snaps')
+
+  const translate = useTranslate()
+  const [shouldAskAgain, toggleShouldAskAgain] = useToggle(true)
+
+  const handleDismiss = useCallback(
+    (_shouldAskAgain: boolean) => {
+      store.dispatch(preferences.actions.setShowSnapssModal(_shouldAskAgain))
+      close()
+    },
+    [close],
+  )
+
+  const allNativeAssets = useMemo(() => {
+    return Object.values(KnownChainIds)
+      .map(knownChainId => {
+        const assetId = getChainAdapterManager().get(knownChainId)?.getFeeAssetId()!
```

Comment:
We may or may not want to filter out EVM chains depending on whether or not they work as fall through with MM snaps in the end - if they don't, we have bigger problems